packages/@aws-cdk/aws-appmesh/lib/virtual-node.ts

```diff
@@ -0,0 +1,220 @@
+  /**
+   * The Amazon Resource Name belonging to the VirtualNdoe
+   *
+   * @type {string}
+   * @memberof VirtualNode
+   */
+  public readonly virtualNodeArn: string;
+
+  private readonly backends: CfnVirtualNode.BackendProperty[] = [];
+  private readonly listeners: CfnVirtualNode.ListenerProperty[] = [];
+  private readonly namespaceName: string;
+
+  constructor(scope: cdk.Construct, id: string, props: VirtualNodeProps) {
+    super(scope, id);
+
+    this.meshName = props.meshName;
+    this.namespaceName = props.namespaceName;
+
+    if (props.backends) {
+      this.addBackends(props.backends);
+    }
+
+    if (props.listeners) {
+      this.addListeners(props.listeners);
+    }
+
+    this.node.apply(new cdk.Tag(NAME_TAG, this.node.path));
+    const name = props.nodeName ? props.nodeName : id;
+
+    const node = new CfnVirtualNode(this, 'VirtualNode', {
+      virtualNodeName: name,
+      meshName: this.meshName,
+      spec: {
+        backends: this.backends,
+        listeners: this.listeners,
+        serviceDiscovery: {
+          dns: {
+            hostname: `${props.hostname}.${this.namespaceName}`,
+          },
+        },
+        logging: {
+          accessLog: {
+            file: {
+              path: '/dev/stdoout',
+            },
+          },
+        },
+      },
+    });
+
+    this.virtualNodeName = node.virtualNodeName;
+    this.virtualNodeArn = node.virtualNodeArn;
+  }
+
+  /**
+   * Utility method to add a single backend for existing or new VritualNodes
+   *
+   * @param {VirtualNodeBackendProps} props
+   * @memberof VirtualNode
+   */
+  public addBackend(props: VirtualNodeBackendProps) {
+    this.backends.push({
+      virtualService: {
+        virtualServiceName: props.virtualServiceName,
+      },
+    });
+  }
+
+  /**
+   * Utility method to add backends for existing or new VritualNodes
+   *
+   * @param {VirtualNodeBackendProps[]} props
+   * @memberof VirtualNode
+   */
+  public addBackends(props: VirtualNodeBackendProps[]) {
+    props.forEach(s => {
+      this.backends.push({
+        virtualService: {
+          virtualServiceName: s.virtualServiceName,
+        },
+      });
+    });
+  }
+
+  /**
+   * Utility method to add Node Listeners for new or existing VirtualNodes
+   *
+   * @param {ListenerProps} props
+   * @memberof VirtualNode
+   */
+  public addListeners(props: ListenerProps) {
+    if (props.portMappings && props.healthChecks) {
+      this.addPortAndHealthCheckMappings(props.portMappings, props.healthChecks);
+    } else if (props.portMappings) {
+      this.addPortMappings(props.portMappings);
+    } else {
+      this.addPortMappings([{ port: 8080, protocol: Protocol.HTTP }]);
+    }
+  }
+
+  /**
+   * Utility method which adds only port mappings to the listener property as healthchecks are optional
+   *
+   * @param {PortMappingProps[]} props
+   * @memberof VirtualNode
+   */
+  public addPortMappings(props: PortMappingProps[]) {
```

Comment:
There seems to be no singular form of this?

Reply:
done, will show in next commit